test/cql-pytest/test_describe.py

```diff
@@ -7,12 +7,591 @@
+                desc = cql.execute(f"DESC KEYSPACE {ks}")
+                ks_desc = " ".join([r.create_statement for r in desc])
+
+                assert f"CREATE KEYSPACE {ks}" in ks_desc
+                assert f"CREATE TYPE {udt}" in ks_desc
+                assert f"CREATE TABLE {tbl}" in ks_desc
+
+                only_desc = cql.execute(f"DESC ONLY KEYSPACE {ks}")
+                ks_only_desc = " ".join([r.create_statement for r in only_desc])
+
+                assert f"CREATE KEYSPACE {ks}" in ks_only_desc
+                assert f"CREATE TYPE {udt}" not in ks_only_desc
+                assert f"CREATE TABLE {tbl}" not in ks_only_desc
+
+# Test that `DESC SCHEMA` contains all informations for user created keyspaces
+# and `DESC FULL SCHEMA` contains also informations for system keyspaces
+def test_desc_schema(cql, test_keyspace, random_seed):
+    with new_random_keyspace(cql) as ks:
+        with new_random_table(cql, test_keyspace) as tbl1, new_random_table(cql, ks) as tbl2:
+            desc = cql.execute("DESC SCHEMA")
+            schema_desc = " ".join([r.create_statement for r in desc])
+
+            assert f"CREATE KEYSPACE {test_keyspace}" in schema_desc
+            assert f"CREATE TABLE {tbl1}" in schema_desc
+            assert f"CREATE KEYSPACE {ks}" in schema_desc
+            assert f"CREATE TABLE {tbl2}" in schema_desc
+
+            full_desc = cql.execute("DESC FULL SCHEMA")
+            schema_full_desc = " ".join([r.create_statement for r in full_desc])
+
+            kss = cql.execute("SELECT keyspace_name FROM system_schema.keyspaces")
+            tbls = cql.execute("SELECT keyspace_name, table_name FROM system_schema.tables")
+
+            for ks_row in kss:
+                assert f"CREATE KEYSPACE {maybe_quote(ks_row.keyspace_name)}" in schema_full_desc
+            for tbl_row in tbls:
+                assert f"CREATE TABLE {maybe_quote(tbl_row.keyspace_name)}.{maybe_quote(tbl_row.table_name)}"
+
+# Test that `DESC CLUSTER` contains token ranges to endpoints map
+# The test is `scylla_only` because there is no `system.token_ring` table in Cassandra
+def test_desc_cluster(scylla_only, cql, test_keyspace):
+    cql.execute(f"USE {test_keyspace}")
+    desc = cql.execute("DESC CLUSTER").one()
+    desc_endpoints = []
+    for (end_token, endpoints) in desc.range_ownership.items():
+        for endpoint in endpoints:
+            desc_endpoints.append((end_token, endpoint))
+
+    result = cql.execute(f"SELECT end_token, endpoint FROM system.token_ring WHERE keyspace_name='{test_keyspace}' ORDER BY start_token")
+    ring_endpoints = [(r.end_token, r.endpoint) for r in result]
+    
+    assert sorted(desc_endpoints) == sorted(ring_endpoints)
+
+def test_keyspaces_quoting(cql):
+    cql.execute("CREATE KEYSPACE \"Quoted_KS\" WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1}")
+
+    try:
+        assert_element_listing(cql, "keyspaces", "keyspace_name", lambda r: r.keyspace_name)
```

Comment:
I'm afraid that this test is too "automatic" for me to verify what it tests, and too uncommented to know what it's supposed to test. The "DESCRIBE KEYSPACES", it returns CQL, right? Does this test check that the CQL command is properly quoting the name Quoted_KS or just that this name appears (which is not enough?)

I have a vague recollection that in the past you wanted to run the CQL command return by DESCRIBE and verify it creates the same table again. If you did that, we could be sure that the CQL command really works, and doesn't result in a syntax error or the wrong table being created.

Reply:
I've described quoting tests in comment now.

These tests don't check if whole create statement is correct (there are other tests for that, above them) but only check if the name is properly quoted and if listing (eg. `DESC TABLES`) also quotes the name/keyspace.